tools/image_processing/yolo/yolo_training.xml

```diff
@@ -0,0 +1,154 @@
+      export YOLO_CONFIG_DIR='\$HOME/.config/ultralytics' &&
+
+      mkdir -p ./input_images ./input_yolo ./training ./runs ./models output_files && 
+
+      #for $i, $filename in enumerate($input_images):
+          ln -s '$filename' './input_images/${filename.element_identifier}' &&
+      #end for
+	
+      #for $s, $yolo in enumerate($input_yolo):
+          ln -s '$yolo' './input_yolo/${yolo.element_identifier}' &&
+      #end for
+
+      python '$__tool_directory__/preprocessing.py' -i './input_images' -y './input_yolo' -o ./training -p '$num_train' -m $input_json &&
+
+      python '$__tool_directory__/download_models.py' -m '$training_params.model_name' -f './models' -c '${model_urls}' &&
+
+      python '$__tool_directory__/yolov8.py' 
+		    --train --yaml_path='./training/yolo.yml' 
+		    --model_path='./models' 
+		    --model_name='$training_params.model_name' 
+		    --run_dir='./runs' 
+		    --image_size='$training_params.image_size'
+		    --epochs='$training_params.epochs'
+		    --scale='$training_params.scale'
+		    --degree='$training_params.degree'
+		    --hsv_v='$training_params.hsv_v'
+		    --hsv_s='$training_params.hsv_s'
+		    --hsv_h='$training_params.hsv_h'
+		    --init_lr='$training_params.init_lr'
+		    --weight_decay='$training_params.weight_decay'
+		    --confidence='$training_params.confidence'
+		    --iou='$training_params.iou'
+		    --max_det='$training_params.max_det'  
+
+      &&
+
+      cp ./runs/train*/*/best.pt output_files/best.pt &&
+      cp ./runs/train*/*/last.pt output_files/last.pt &&
+      cp ./runs/train*/results.csv output_files/results_metrics.csv &&
+      cp ./runs/train*/results.png output_files/results_plot.png 
+    ]]>
+    </command>
+    <configfiles>
+	 <inputs name="input_json"/>
+	 <configfile name="model_urls"><![CDATA[
+            {
+		 "yolo11n-seg": "https://github.com/ultralytics/assets/releases/download/v8.3.0/yolo11n-seg.pt",
+		 "yolo11n": "https://github.com/ultralytics/assets/releases/download/v8.3.0/yolo11n.pt",
+		 "yolov8n": "https://github.com/ultralytics/assets/releases/download/v8.3.0/yolov8n.pt",
+		 "sam_b": "https://github.com/ultralytics/assets/releases/download/v8.3.0/sam_b.pt",
+		 "sam_l": "https://github.com/ultralytics/assets/releases/download/v8.3.0/sam_l.pt",
+		 "sam2_t": "https://github.com/ultralytics/assets/releases/download/v8.3.0/sam2_t.pt",
+		 "sam2.1_t": "https://github.com/ultralytics/assets/releases/download/v8.3.0/sam2.1_t.pt"
+            }
+            ]]>
+	 </configfile>
+    </configfiles>
+    <inputs>
+        <param name="input_images" type="data" format="jpg,png,tiff" multiple="true"  label="Input images"/>
+        <param name="input_yolo" type="data" format="txt" multiple="true"  label="Input YOLO txt files" help="The YOLO text files, each text file must correspond to one input image(same name different extension)." />
+	<param name="ds_name" type="text" label="Name of dataset" help="The name for your dataset"/>
```

Comment:
What is this parameter doing?

Reply:
It is required in yolo.yaml which is created in python and needed during training

Reply:
Can we hordcode something? Will the user ever see this? If it is really needed from a user I think we need to change the label or help section to describe what it is, especially because we deal with many files/datasets here.